### PR TITLE
Be able to return JSON data in checker

### PR DESCRIPTION
**Pull request by sbrunner**
Example of usage:
https://github.com/camptocamp/c2cgeoportal/pull/4258

### CODE REVIEW

File: c2cwsgiutils/health_check.py [hunk 1]
```diff
@@ -18,14 +18,34 @@
...
-from typing import Optional, Callable, Mapping, List, Tuple, Any, Union
+from typing import Optional, Callable, Mapping, List, Tuple, Any, Union, Dict  # noqa
 
 from c2cwsgiutils import stats, _utils, broadcast
 
 LOG = logging.getLogger(__name__)
 ALEMBIC_HEAD_RE = re.compile(r'^([a-f0-9]+) \(head\)\n$')
 
 
+class JsonCheckException(Exception):
```
Comment (pvalsecc) [inline c2cwsgiutils/health_check.py]: Since it's public, please document the purpose of this class.

File: c2cwsgiutils/health_check.py [hunk 2]
```diff
@@ -221,6 +241,15 @@ def _view(self, request: pyramid.request.Request) -> Mapping[str, Any]:
                     }
                     if result is not None:
                         results['successes'][name]['result'] = result
+                except JsonCheckException as e:
+                    LOG.warning("Health check %s failed", name, exc_info=True)
+                    failure = {
+                        'timing': time.monotonic() - start
```
Comment (pvalsecc) [inline c2cwsgiutils/health_check.py]: Missing the message.

File: c2cwsgiutils/health_check.py [hunk 3]
```diff
@@ -221,6 +241,15 @@ def _view(self, request: pyramid.request.Request) -> Mapping[str, Any]:
...
+                    }  # type: Dict[str, Any]
+                    failure.update(e.data())
+                    if os.environ.get('DEVELOPMENT', '0') != '0':
+                        failure['stacktrace'] = traceback.format_exc()
+                    results['failures'][name] = failure
```
Comment (pvalsecc) [inline c2cwsgiutils/health_check.py]: This is a lot of copy&paste. Please refactor into a function or just use a `if instanceof(...` for the short part about JsonCheckException.

File: c2cwsgiutils/health_check.py (hunk 2 above)
Comment (sbrunner) [reply c2cwsgiutils/health_check.py]: Added in the next line...

**Pull request merged**